### PR TITLE
Updating vtrutil api doc

### PR DESCRIPTION
**Pull request by Srivat97**
Added new information to the VTRUTIL API documentation. The updates improved the documentation clarity.

#### Description

- First update was making it clear that VTR_LOG_ERROR does not terminate the program
- Second update was adding an example on how vtr::VtrError is used, and also indicated that it can be used to terminate the program.

#### Motivation and Context

- These changes were made to help improve the VTR documentation.

#### How Has This Been Tested?

- After the changes were made, the documentation was built and the changes were verified visually. 

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: libs/libvtrutil/src/vtr_error.h [hunk 1]
```diff
@@ -4,15 +4,32 @@
 #include <stdexcept>
 #include <string>
 
+/**
+ * @file
+ * @brief A utility container that can be used to identify VTR execution errors.
+ * 
+ * The recommended usage is to store error information within this container during an error event and handling the error with exceptions. If the exception is not handled (exception is not caught), this will result in the termination of the program.
```
Comment (vaughnbetz) [inline libs/libvtrutil/src/vtr_error.h:11]: "handling with exceptions" -> "store information in this container about the error, and then throw an exception with the container"

File: libs/libvtrutil/src/vtr_log.h [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * Three types of log message types are defined:
  *     - VTR_LOG         : The standard 'info' type log message
  *     - VTR_LOG_WARN    : A warning log message. This represents unusual condition that may indicate an issue but executiom continues
- *     - VTR_LOG_ERROR   : An error log message. This represents a clear issue that will result in stopping the program execution
+ *     - VTR_LOG_ERROR   : An error log message. This represents a clear issue that should result in stopping the program execution. Pease note that using this log message will not actually terminate the program
```
Comment (vaughnbetz) [inline libs/libvtrutil/src/vtr_log.h:17]: Add a period at end of sentence. 
Pease -> Please
Add: "so a VtrError should be thrown after all the VTR_LOG_ERROR messages are printed."

**Comment by vaughnbetz** [comment]
A few tweaks suggested above.

File: libs/libvtrutil/src/vtr_error.h (hunk 1 above)
Comment (Srivat97) [reply libs/libvtrutil/src/vtr_error.h]: Added the requested change

File: libs/libvtrutil/src/vtr_log.h (hunk 2 above)
Comment (Srivat97) [reply libs/libvtrutil/src/vtr_log.h]: Added the requested change

**Comment by Srivat97** [comment]
The requested changed to the initial review have all been added to the documentation.

**Pull request merged**